src/pcrlock/pcrlock.c

```diff
@@ -4567,15 +4567,58 @@ static int make_policy(bool force, bool recovery_pin) {
                 log_info("Retrieved PIN from TPM2 in %s.", FORMAT_TIMESPAN(usec_sub_unsigned(now(CLOCK_MONOTONIC), pin_start_usec), 1));
         }
 
-        TPM2B_NV_PUBLIC nv_public = {};
+        /* Load the PIN (which we have stored in the "auth" TPM2B_AUTH) into the TPM as an HMAC key so that
+         * we can use it in a TPM2_PolicySigned() to write to the nvindex */
+        _cleanup_(tpm2_handle_freep) Tpm2Handle *pin_handle = NULL;
+
+        TPM2B_PUBLIC auth_hmac_public = {
+                .size = sizeof(TPMT_PUBLIC),
+                .publicArea = {
+                        .type = TPM2_ALG_KEYEDHASH,
+                        .nameAlg = TPM2_ALG_SHA256,
+                        .objectAttributes = TPMA_OBJECT_DECRYPT | TPMA_OBJECT_SIGN_ENCRYPT /* | TPMA_OBJECT_USERWITHAUTH */,
+                        .parameters.keyedHashDetail.scheme.scheme = TPM2_ALG_SHA256,
```

Comment:
The object has no way to be used, probably want that userwithauth attribute. It either needs a policy or userwithauth unless its used by the admin (that may have an attribute, I cannot recall offhand), which is only certain commands.

Reply:
This is being used in `TPM2_PolicySigned` which doesn't require any auth with the object - I think the lack of `userWithAuth` here is ok.

Reply:
My point is it makes no sense to set the private objects passphrase without this attribute. Also without a policy, the loaded object is only useable by commands that don't require authorization. If this object has intentions to be used later where auth is needed, that attribute would be needed.

It's either a set both things or don't.